core/dbt/parser/schemas.py

```diff
@@ -936,59 +938,41 @@ def parse_patch(self, block: TargetBlock[NodeTarget], refs: ParserRef) -> None:
                 self.patch_node_config(node, patch)
 
             node.patch(patch)
-            self.validate_constraints(node)
-
-    def validate_constraints(self, patched_node):
-        error_messages = []
-        if patched_node.resource_type == "model" and patched_node.config.contract is True:
-            validators = [
-                self.constraints_schema_validator(patched_node),
-                self.constraints_materialization_validator(patched_node),
-                self.constraints_language_validator(patched_node),
-            ]
-            error_messages = [validator for validator in validators if validator != "None"]
-
-        if error_messages:
-            original_file_path = patched_node.original_file_path
-            raise ParsingError(
-                f"Original File Path: ({original_file_path})\nConstraints must be defined in a `yml` schema configuration file like `schema.yml`.\nOnly the SQL table and view materializations are supported for constraints. \n`data_type` values must be defined for all columns and NOT be null or blank.{self.convert_errors_to_string(error_messages)}"
+
+            if isinstance(node, ModelNode) and node.config.contract is True:
```

Comment:
This code works, but right now we're doing node-level field setting in the patch method so ideally this should happen in that method if possible.  I presume that access to "validate_constraints" is the issue? Can that be changed into a function that can be called in both places?

Reply:
Alternatively, we could pull that "patch" method into where this code is called from. It doesn't need to stay on the node object...

Reply:
Per huddle convo, I've added some comments for now and we'll plan to make this more consistent soon.